### DISCUSSION TITLE
Data migration deadline?

### DISCUSSION THREAD

**Discussion by singularity-s0** [OpenTreeHole/backend#16] [General]
To prevent infinite delay of development, @w568w and I believe it is necessary that we set a deadline for the Data Migration Project.

Taking all the technical difficulties into consideration, @hasbai what deadline do you think is appropriate for us?

**Comment by hasbai** [comment]
由于新版本的 API 有很大变动，后端更新后新的前端也需要立即上线，因此需要等到所有的前端项目都完成后才能开始更新工作，因此无法给出一个截止日期。
考虑到目前旧版本仍能稳定运行，我将优先完成新版本的开发工作，并解决部署运维上的一系列问题。如果你们目前的工作已经完成，欢迎积极参与其它未完成项目的开发。

**Comment by singularity-s0** [comment]
我认为此问题应当优先处理。

1. 数据迁移是新版本上线的基础。非关键的功能可以后续再添加，但数据迁移若无法实现则新版本始终不能上线。
1. 虽然API尚未最终确定，但目前至少应该实现一个雏形。这样即使之后数据库结构又有变动，改动所需的工作量也非常小。

如果不熟悉相关工作，可以指定一位或几位熟悉的同学协助你开发。但**无限期拖延**是不应该的。

**Comment by hasbai** [comment]
请注意，目前的 web 前端项目尚未完成，在其完成之前，新版本都无法投入使用，因此数据迁移的工作并不是当务之急。
